### ISSUE TITLE
track_script directive since version 2.1.5

### ISSUE THREAD

**Issue by gunhu** [acassen/keepalived#1813]
**Describe the bug**
I'm using keepalived (2.1.5) with conntrackd daemon to synchronize firewall states.

**To Reproduce**
Here is the configuration between the master and backup nodes:

**MASTER node**
```
# Please do not change this file directly since it is managed by Ansible and will be overwritten

vrrp_script enable_maintenance {
  script "/etc/keepalived/vrrp_script/enable_maintenance.sh"
  interval 2
  fall 1
  rise 1
  }

vrrp_sync_group VG1 {
  group {
      lan_prod
      wan
    }
    notify_master "/etc/conntrackd/primary-backup.sh primary"
    notify_backup "/etc/conntrackd/primary-backup.sh backup"
    notify_fault "/etc/conntrackd/primary-backup.sh fault"
}

vrrp_instance lan_prod {
  interface bond0.999
  virtual_router_id 122
  priority 100
  authentication {
    auth_type PASS
    auth_pass secret_password
  }
  virtual_ipaddress {
    10.101.0.1
  }
  track_script {
    enable_maintenance
  }
  track_interface {
    bond0
    bond0.999
  }

}
vrrp_instance wan {
  interface bond0.101
  virtual_router_id 122
  priority 100
  authentication {
    auth_type PASS
    auth_pass secret_password
  }
  virtual_ipaddress {
    1.1.1.1
  }
  track_script {
    enable_maintenance
  }
  track_interface {
    bond0
    bond0.101
  }

}
```
**BACKUP node**
```
# Please do not change this file directly since it is managed by Ansible and will be overwritten

vrrp_script enable_maintenance {
  script "/etc/keepalived/vrrp_script/enable_maintenance.sh"
  interval 2
  fall 1
  rise 1
  }

vrrp_sync_group VG1 {
  group {
      lan_prod
      wan
    }
    notify_master "/etc/conntrackd/primary-backup.sh primary"
    notify_backup "/etc/conntrackd/primary-backup.sh backup"
    notify_fault "/etc/conntrackd/primary-backup.sh fault"
}

vrrp_instance lan_prod {
  interface bond0.999
  virtual_router_id 122
  priority 100
  authentication {
    auth_type PASS
    auth_pass secret_password
  }
  virtual_ipaddress {
    10.101.0.1
  }
  track_script {
    enable_maintenance
  }
  track_interface {
    bond0
    bond0.999
  }

}
vrrp_instance wan {
  interface bond0.101
  virtual_router_id 122
  priority 100
  authentication {
    auth_type PASS
    auth_pass secret_password
  }
  virtual_ipaddress {
    1.1.1.1
  }
  track_script {
    enable_maintenance
  }
  track_interface {
    bond0
    bond0.101
  }

}
```
**Expected behavior**
Track script working like before, instead I got the following warning message:
```
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Warning - script enable_maintenance is not used
```

**Keepalived version**
Output of `keepalived -v`

**Distro:**
 - Debian
 - Buster (10)
 - x86_64

**Notify and track scripts**
```
#!/bin/bash

if [ -f /etc/keepalived/enable_maintenance ]; then
  logger -t Keepalived "Shhhh.... We are now sleeping and dreaming about new features to implement. Will be back soon."
  exit 1
fi
```
**System Log entries**
```
Dec 14 14:19:49 natgw-01 Keepalived[10826]: Starting Keepalived v2.1.5 (unknown)
Dec 14 14:19:49 natgw-01 Keepalived[10826]: WARNING - keepalived was build for newer Linux 4.19.152, running on Linux 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18)
Dec 14 14:19:49 natgw-01 Keepalived[10826]: Command line: '/usr/sbin/keepalived'
Dec 14 14:19:49 natgw-01 Keepalived[10826]: Opening file '/etc/keepalived/keepalived.conf'.
Dec 14 14:19:49 natgw-01 Keepalived[10827]: NOTICE: setting config option max_auto_priority should result in better keepalived performance
Dec 14 14:19:49 natgw-01 Keepalived[10827]: Starting VRRP child process, pid=10828
Dec 14 14:19:49 natgw-01 systemd[1]: Started Keepalive Daemon (LVS and VRRP).
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Registering Kernel netlink reflector
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Registering Kernel netlink command channel
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Opening file '/etc/keepalived/keepalived.conf'.
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: WARNING - default user 'keepalived_script' for script execution does not exist - please create.
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: SECURITY VIOLATION - scripts are being executed but script_security not enabled.
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: (lan_prod) Ignoring track_interface bond0.999 since own interface
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: (wan) Ignoring track_interface bond0.101 since own interface
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Warning - script enable_maintenance is not used
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: Registering gratuitous ARP shared channel
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: (lan_prod) Entering BACKUP STATE (init)
Dec 14 14:19:49 natgw-01 Keepalived_vrrp[10828]: (wan) Entering BACKUP STATE (init)
```
**Did keepalived coredump?**
No.

**Comment by gunhu** [comment]
The fix here is to declare the track_script in the vrrp_sync_group block instead.

Is it the new way to do so when we use conntrack?

Regards,

**Comment by gunhu** [comment]
I tested successfully the same configuration with 2.0.19 without issues.

Breaking occurs only with 2.1.* releases.

**Comment by pqarmitage** [comment]
This was resolved by commit 459fc2b. The patch applies cleaning to v2.1.5, or you could just use the latest code for github.

**Comment by pqarmitage** [comment]
@gunhu I have noticed the following errors in your configurations:

1. The VRRP instances should have different priorities on the different nodes
2. The log messages report:
```
(lan_prod) Ignoring track_interface bond0.999 since own interface
(wan) Ignoring track_interface bond0.101 since own interface
```
These interfaces should be removed from the `track_interfaces` blocks.
